generator/templates/Default/src/application/models/model.js

```diff
@@ -12,10 +11,66 @@ class Model {
 
   /**
    * @param data {{}}
+   * @return {model}
    */
   mapResponse(data) {
-    throw new Error(`${this.constructor.name}::mapResponse unset, keys: ${Object.keys(data)
-      .join(', ')}`);
+    return this.initializeObjectFromApi(data, this);
+  }
+
+  /**
+   * @private
+   * @param data
+   * @param assignable
+   * @return {model}
+   */
+  initializeObjectFromApi(data, assignable) {
+    Object.keys(assignable)
+      .forEach(key => {
+        const value = assignable[key];
+        const type = typeof value;
+
+        if (typeof data[key] === type) {
+          assignable[key] = data[key];
+          return;
+        }
+
+        const defaultValue = this.getDefaultValueFromType(type, value);
+
+        assignable[key] = data[key] || defaultValue;
+      });
+
+    Object.keys(data)
+      .forEach(key => {
+        if (typeof assignable[key] === 'undefined') {
+          assignable[key] = data[key];
+        }
+      });
+
+    return assignable;
+  }
+
+  /**
+   * @private
+   * @param type
+   * @param defaultModelValue
+   * @return {*}
+   */
+  getDefaultValueFromType(type, defaultModelValue) {
+    let defaultValue;
+
+    if (type === 'string') {
```

Comment:
switch case?

Reply:
There wouldn't be a difference, because each case would be different in this case. And not every value being checked is of type string (the array function call)